syntax = "proto3";
package commerce_webhook;

import "google/protobuf/struct.proto";
import "google/protobuf/timestamp.proto";

service CommerceWebhookService {
    rpc InsertCommerceWebhookData (InsertCommerceWebhookRequest) returns (InsertCommerceWebhookResponse);
}

message InsertCommerceWebhookRequest {
    int32 status = 1;
    string template = 2;
    google.protobuf.Struct template_variables = 3;
    string contact_urn = 4;
    google.protobuf.Struct error = 5;
    google.protobuf.Struct data = 6;
    google.protobuf.Timestamp date = 7;
    string project = 8; // UUID as string
}

message InsertCommerceWebhookResponse {
    string status = 1;
} 